pkg/util/mapper/mapper.go

```diff
@@ -464,8 +486,74 @@ func (wom *writeOnlyMapper) SetBool(key string, value bool) error {
 	return wom.setter.Set(key, strconv.FormatBool(value))
 }
 
+// SetDuration sets the map to a string formatted bool value.
+func (wom *writeOnlyMapper) SetDuration(key string, value time.Duration) error {
+	return wom.setter.Set(key, durationString(value))
+}
+
 // SetList sets the map's value at key to a string consistent of each value in the list separated
 // by the provided delimiter.
 func (wom *writeOnlyMapper) SetList(key string, values []string, delimiter string) error {
 	return wom.setter.Set(key, strings.Join(values, delimiter))
 }
+
+const (
+	secsPerMin  = 60.0
+	secsPerHour = 3600.0
+	secsPerDay  = 86400.0
+)
+
```

Comment:
```suggestion

// durationString converts duration to a string of the form "4d", "4h", "4m", or "4s" if
// the number of seconds in the string is evenly divisible into an integer number of
// days, hours, minutes, or seconds respectively.
```

Reply:
Yep, nice. (Unfortunately, Bolt and I decided that copy-pasting this function was the best way to fix a circular dependency... so annoying to have it here at all, but everything else we thought to do was worse.)